drivers/spi/spi_mcux_flexcomm.c

```diff
@@ -631,12 +631,19 @@ static int transceive_dma(const struct device *dev,
 
 		data->status_flags = 0;
 
+		if (data->ctx.rx_buf == NULL) {
+			dummy_rx_buffer = k_malloc(dma_len);
```

Comment:
Shouldn't this not be needed if the DMA is setup to not increment the address of the receive when using the dummy word?

Reply:
Great point @teburd and thank you for your review comment.
On further investigation I did come across for a similar feature in this controller as well. I have updated the PR.